runtime/flang/CMakeLists.txt

```diff
@@ -477,54 +477,126 @@ SET(FTN_SUPPORT
   xfer_rpm1.c
   )
 
-add_library(iso_c_bind OBJECT
-  iso_c_bind.F95
-)
+foreach(FTN_FILE ieee_arithmetic ieee_exceptions iso_c_bind)
+  configure_file(${FTN_FILE}.F95 ${FTN_FILE}_static.F95 COPYONLY)
+endforeach()
 
-add_library(ieee_arithmetic OBJECT
+SET(FTN_SUPPORT_SHARED
   ieee_arithmetic.F95
   ieee_exceptions.F95
+  iso_c_bind.F95
+)
+
+SET(FTN_SUPPORT_STATIC
+  ieee_arithmetic_static.F95
+  ieee_exceptions_static.F95
+  iso_c_bind_static.F95
+)
+
+SET(FTN_SUPPORT
+  ${FTN_SUPPORT_STATIC}
+  ${FTN_SUPPORT_SHARED}
+  ${FTN_SUPPORT_COMMON}
 )
 
 add_flang_library(flang_static
   ${FTN_INTRINSICS}
-  ${FTN_SUPPORT}
+  ${FTN_SUPPORT_COMMON}
+  ${FTN_SUPPORT_STATIC}
   ${SHARED_SOURCES}
-  $<TARGET_OBJECTS:iso_c_bind>
-  $<TARGET_OBJECTS:ieee_arithmetic>
   )
   
-set_target_properties(flang_static
-  PROPERTIES
-  Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/include-static
-)
-
 if (MSVC)
   set_property(TARGET flang_static PROPERTY OUTPUT_NAME libflang)
 else()
   set_property(TARGET flang_static PROPERTY OUTPUT_NAME flang)
 endif()
 
+set_target_properties(flang_static PROPERTIES Fortran_MODULE_DIRECTORY ${FLANG_MODULE_DIRECTORY_STATIC})
+
+if (NOT "${CMAKE_GENERATOR}" STREQUAL "Ninja")
+# State the module that the source is producing
+set_source_files_properties(
+  iso_c_bind_static.F95
+  PROPERTIES
+  OBJECT_OUTPUTS ${FLANG_MODULE_DIRECTORY_STATIC}/iso_c_binding.mod
+  )
+
+# State a dependency on the module
+set_source_files_properties(
+  ieee_arithmetic_static.F95
+  ieee_exceptions_static.F95
+  PROPERTIES
+  OBJECT_DEPENDS ${FLANG_MODULE_DIRECTORY_STATIC}/iso_c_binding.mod
+  )
+else()
+add_library(iso_c_bind_static OBJECT
```

Comment:
I now have two object libraries. `iso_c_bind_static` and `iso_c_bind_shared` which should fix the issues we were having.